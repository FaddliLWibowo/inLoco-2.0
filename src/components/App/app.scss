#app, .module-app {
    height: 100%;
}

@import "sass/settings.scss";
@import "sass/tools.scss";
@import "sass/generic.scss";
@import "sass/components.scss";
@import "sass/animations.scss";
