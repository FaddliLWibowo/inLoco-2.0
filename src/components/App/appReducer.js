import geoServerXmlReducer from './reducers/geoServerXmlReducer'
import menuReducer from '../Menu/menuReducer'

const appReducer = (state = [], action) => {
    switch(action.type){
        case 'POPULATE_APP':
            let layers = geoServerXmlReducer(action.xmlData.xmlData);
            layers = layers.map(l => {
                return {
                    ...l,
                    selected: false,
                    match: true,
                    showDescription: false
                }
            });
            let menuItems = menuReducer(layers)
            menuItems = menuItems.map(m => {
                return {
                    ...m,
                    selected: false,
                    match: true,
                }
            });
            let tooltip = {
                text: "",
                show: false,
                y: undefined,
                sidebarLeftWidth: 0,
                sidebarLeftHeight: 0
            }
            return {
                currentLevel: 0,
                layers,
                menuItems,
                showMenu: false,
                tooltip
            };
        case 'TOGGLE_LAYER':
            var newLayers = []
            newLayers = state.layers.map(l => layer(l, action))
            return {
                currentLevel: state.currentLevel,
                layers: newLayers,
                menuItems: state.menuItems,
                showMenu: state.showMenu,
                tooltip: state.tooltip
            };
        case 'TOGGLE_MENU':
            let currentLevel = state.currentLevel
            var newLayers = []

            // make a copy of the state
            var newMenuItems = [...state.menuItems]

            // if i'm closing
            if (action.selected === true) {
                // find myself
                newMenuItems.forEach(menuItem => {
                    if (menuItem.id === action.id) {
                        // if i have children
                        if (menuItem.submenus.length > 0) {
                            // close my children
                            menuItem.submenus.forEach(submenu => {
                                // find this submenu in menus array
                                newMenuItems.forEach(thisMenuItem => {
                                    if (thisMenuItem.idMenu === submenu) {
                                        // close it
                                        thisMenuItem.selected = false
                                    }
                                })
                            })
                        } else {
                            // i don't have children, close myself
                            newMenuItems = newMenuItems.map(m => menuItem(m, action, state.currentLevel))
                        }
                    }
                })
            } else {
                // if i'm opening, do it right away
                newMenuItems = newMenuItems.map(m => menuItem(m, action, state.currentLevel))
            }
            if (action.selected) {
                currentLevel--
            } else {
                currentLevel++
            }
            return {
                currentLevel,
                layers: state.layers,
                menuItems: newMenuItems,
                showMenu: state.showMenu,
                tooltip: state.tooltip
            };
        case 'UNTOGGLE_MENUS':
            var newMenuItems = state.menuItems.map(m => menuItem(m, action))
            return {
                currentLevel: 0,
                layers: state.layers,
                menuItems: newMenuItems,
                showMenu: state.showMenu,
                tooltip: state.tooltip
            }
        case 'SEARCH_LAYER':
            var newLayers = state.layers.map(l => searchLayer(l, action))
            var filteredLayers = newLayers.filter(layer => layer.match)
            var newMenuItems = []
            if(action.text === ''){
                // when emptying search, return all items
                newMenuItems = state.menuItems.map(m => {
                    return {
                        ...m,
                        match: true,
                    }
                })
            } else {
                newMenuItems = state.menuItems.map(m => searchMenuItem(m, filteredLayers))
            }

            return {
                currentLevel: state.currentLevel,
                layers: newLayers,
                menuItems: newMenuItems,
                showMenu: state.showMenu,
                tooltip: state.tooltip
            }
        case 'SHOW_DESCRIPTION':
            var layerResult = state.layers.find(l => layer(l, action));
            var newTooltip;
            if(layerResult){
                newTooltip = {
                    text: layerResult.description,
                    show: true,
                    y: action.y,
                    sidebarLeftWidth: action.sidebarLeftWidth,
                    sidebarLeftHeight: action.sidebarLeftHeight
                }
            } else {
                newTooltip = {
                    text: "",
                    show: false
                }
            }
            return {
                currentLevel: state.currentLevel,
                layers: state.layers,
                menuItems: state.menuItems,
                showMenu: state.showMenu,
                tooltip: newTooltip
            }
        case 'HIDE_DESCRIPTION':
            return {
                currentLevel: state.currentLevel,
                layers: state.layers,
                menuItems: state.menuItems,
                showMenu: state.showMenu,
                tooltip: {
                    text: "",
                    show: false
                }
            }
        case 'SHOW_MENU_LAYER':
            return {
                currentLevel: state.currentLevel,
                layers: state.layers,
                menuItems: state.menuItems,
                showMenu: true,
                tooltip: state.tooltip
            }
        case 'HIDE_MENU_LAYER':
            return {
                currentLevel: state.currentLevel,
                layers: state.layers,
                menuItems: state.menuItems,
                showMenu: false,
                tooltip: state.tooltip
            }
        default:
            return state
    }
}

const layer = (layer, action) => {
    switch (action.type){
        case('TOGGLE_LAYER'):
            if (layer.id !== action.id) {
                return layer
            }
            return {
                ...layer,
                selected: !layer.selected,
            }
        case('TOGGLE_MENU'):
            if (layer.id !== action.id){
                return layer
            }
            return {
                ...layer,
                selected: !layer.match,
            }
        case('SHOW_DESCRIPTION'):
            if (layer.id === action.id) {
                return layer
            }
            return undefined;
        case('HIDE_DESCRIPTION'):
            if (layer.id !== action.id) {
                return layer
            }
            return {
                ...layer,
                showDescription: false,
            };
        default:
            return layer
    }
}

const menuItem = (menuItem, action, currentLevel) => {
    switch (action.type){
        case('TOGGLE_MENU'):
            if (menuItem.id !== action.id) {
                return menuItem
            }
            return {
                ...menuItem,
                selected: !menuItem.selected,
            }
        case 'UNTOGGLE_MENUS':
            return {
                ...menuItem,
                match: true,
                selected: false,
            }
        default:
            return menuItem
    }
}

const searchMenuItem = (menuItem, layers) => {
    var layerMatch = false
    menuItem.layers.forEach(function(menuItemLayer) {
        layers.forEach(function(layer) {
            if (layer.key === menuItemLayer) {
                layerMatch = true
            }
        })
    })
    if(layerMatch){
        return {
            ...menuItem,
            match: true,
        }
    } else {
        return {
            ...menuItem,
            match: false,
        }
    }
}

const searchLayer = (layer, action) => {
    if (action.text === "") {
        return {
            ...layer,
            match: true,
        }
    }

    if (layer.title.toLowerCase().includes(action.text.toLowerCase())) {
        return {
            ...layer,
            match: true,
        }
    } else if (layer.description.toLowerCase().includes(action.text.toLowerCase())) {
        return {
            ...layer,
            match: true,
        }
    } else {
        return {
            ...layer,
            match: false,
        }
    }
}

export default appReducer
