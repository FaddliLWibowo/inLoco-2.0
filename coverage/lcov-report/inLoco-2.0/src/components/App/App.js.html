<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for inLoco-2.0/src/components/App/App.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">inLoco-2.0/src/components/App</a> App.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import Input from '../input/Input.js';
import ExampleHighcharts from '../Charts/ExampleHighcharts.js';
&nbsp;
require('./app.scss');
&nbsp;
/**
 * Componente App que representa a aplicação
 */
&nbsp;
export default class App extends React.Component {
&nbsp;
    /**
     * construtor do elemento
     */
&nbsp;
    constructor() {
        super();
        this.options = {/*Highcharts options*/}
    }
&nbsp;
    /**
     * renderiza o elemento
     * @return html de marcação do elemento
     */
&nbsp;
    render() {
        return (
            &lt;div style={{textAlign: 'center'}} className="module-app"&gt;
                &lt;h1&gt;Hello World 30&lt;/h1&gt;
                &lt;Input /&gt;
                &lt;hr /&gt;
            &lt;/div&gt;
        );
    }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Apr 07 2017 18:39:23 GMT-0300 (Hora oficial do Brasil)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
