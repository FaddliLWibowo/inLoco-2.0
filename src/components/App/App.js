import React from 'react';
import ReactDOM from 'react-dom';
import logger from 'redux-logger'
import { applyMiddleware, createStore } from 'redux';
import appReducer from './appReducer.js';
import { Provider } from 'react-redux';
import LeafletMapContainer from '../LeafletMap/LeafletMapContainer.js';
import SidebarLeftContainer from '../SidebarLeft/SidebarLeftContainer.js';
import HeaderContainer from '../Header/HeaderContainer.js';
import ExampleHighcharts from '../Charts/ExampleHighcharts.js';
import GeoAPI from '../Api/GeoAPI.js';
import TooltipContainer from '../Tooltip/TooltipContainer.js'
import { populateApp } from '../../actions/actions.js';

require('./app.scss');

const store = createStore(appReducer, applyMiddleware(logger));

const ajaxCallback = (xmlData) => {
    store.dispatch(populateApp(xmlData));
};
GeoAPI.getContent(ajaxCallback);

const App = () => {
    return (
         <Provider store={store}>
            <div className="module-app">
                <HeaderContainer/>
                <TooltipContainer />
                <SidebarLeftContainer />
                <LeafletMapContainer />
            </div>
        </Provider>
    );
}

ReactDOM.render(
    <App/>, document.getElementById('app')
);
