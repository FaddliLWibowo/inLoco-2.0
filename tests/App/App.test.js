import React from 'react';
import App from '../../src/components/App/App.js';
import renderer from 'react-test-renderer';

it('componente renderiza corretamente', () => {
    const app  = shallow(<App/>);
    expect(app).toMatchSnapshot();
});
