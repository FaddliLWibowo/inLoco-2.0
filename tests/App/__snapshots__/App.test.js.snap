// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componente renderiza corretamente 1`] = `
<div
  className="module-app"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Hello World 30
  </h1>
  <Input />
  <hr />
</div>
`;
